embassy-net/src/udp.rs

```diff
@@ -26,13 +25,13 @@ pub enum Error {
     NoRoute,
 }
 
-pub struct UdpSocket<'a> {
-    stack: &'a RefCell<SocketStack>,
+pub struct UdpSocket<'a, D: Driver> {
```

Comment:
It is intentional that the socket is not generic on the driver. It prevents generics spam proliferation.